src/Objects/KeyboardLayout.vala

```diff
@@ -246,79 +246,31 @@ public class Installer.KeyboardLayout : GLib.Object {
         return a.display_name.collate (b.display_name);
     }
 
-    private const string XKB_RULES_FILE = "base.xml";
-
-    private static string get_xml_rules_file_path () {
-        unowned string? base_path = GLib.Environment.get_variable ("XKB_CONFIG_ROOT");
-        if (base_path == null) {
-            base_path = Build.XKB_BASE;
-        }
-
-        return Path.build_filename (base_path, "rules", XKB_RULES_FILE);
-    }
-
     public static GLib.ListStore get_all () {
         var layout_store = new GLib.ListStore (typeof (KeyboardLayout));
-        unowned Xml.Doc* doc = Xml.Parser.read_file (get_xml_rules_file_path ());
-
-        Xml.XPath.Context cntx = new Xml.XPath.Context (doc);
-        unowned Xml.XPath.Object* res = cntx.eval_expression ("/xkbConfigRegistry/layoutList/layout");
-        if (res == null) {
-            delete doc;
-            critical ("Unable to parse 'base.xml'");
-            return layout_store;
-        }
 
-        if (res->type != Xml.XPath.ObjectType.NODESET || res->nodesetval == null) {
-            delete res;
-            delete doc;
-            critical ("No layouts found in 'base.xml'");
-            return layout_store;
-        }
+        var xkb_context = new Rxkb.Context (NO_FLAGS);
+        xkb_context.parse_default_ruleset ();
 
-        for (int i = 0; i < res->nodesetval->length (); i++) {
-            unowned Xml.Node* layout_node = res->nodesetval->item (i);
-            unowned Xml.Node* config_node = get_xml_node_by_name (layout_node, "configItem");
-            unowned Xml.Node* variant_node = get_xml_node_by_name (layout_node, "variantList");
-            unowned Xml.Node* description_node = get_xml_node_by_name (config_node, "description");
-            unowned Xml.Node* name_node = get_xml_node_by_name (config_node, "name");
-            if (name_node == null || description_node == null) {
+        var xkb_layout = xkb_context.get_first_layout ();
+        while (xkb_layout != null) {
+            if (xkb_layout.get_variant () != null) {
                 continue;
             }
```

Comment:
This looks like an infinite loop if `xkb_layout.get_variant () != null` because the value of `xkb_layout` wouldn't change here

Reply:
Ah yeah that was just kind of leftovers from trying to figure out how to handle variants. Thanks for catching that!